contrib/resource_util/src/main/java/io/opencensus/contrib/resource/util/K8sResource.java

```diff
@@ -65,20 +76,47 @@
    * @param podName the k8s pod name.
    * @return a {@link Resource} that describes a k8s container.
    * @since 0.20
+   * @deprecated in favor of {@link #create(String, String, String, String)}.
    */
+  @Deprecated
   public static Resource create(String clusterName, String namespace, String podName) {
+    return create(clusterName, namespace, podName, "");
+  }
+
+  /**
+   * Returns a {@link Resource} that describes Kubernetes deployment service.
+   *
+   * @param clusterName the k8s cluster name.
+   * @param namespace the k8s namespace.
+   * @param podName the k8s pod name.
+   * @param deploymentName the k8s deployment name.
+   * @return a {@link Resource} that describes a k8s container.
+   * @since 0.24
+   */
+  public static Resource create(
+      String clusterName, String namespace, String podName, String deploymentName) {
     Map<String, String> labels = new LinkedHashMap<String, String>();
     labels.put(CLUSTER_NAME_KEY, checkNotNull(clusterName, "clusterName"));
     labels.put(NAMESPACE_NAME_KEY, checkNotNull(namespace, "namespace"));
     labels.put(POD_NAME_KEY, checkNotNull(podName, "podName"));
+    labels.put(DEPLOYMENT_NAME_KEY, checkNotNull(deploymentName, "deploymentName"));
     return Resource.create(TYPE, labels);
   }
 
   static Resource detect() {
+    String podName = firstNonNull(System.getenv("HOSTNAME"), "");
+    String deploymentName = "";
+    // Extract deployment name from the pod name. Pod name is created using
+    // format: [deployment-name]-[Random-String-For-ReplicaSet]-[Random-String-For-Pod]
+    List<String> parts = splitter.splitToList(podName);
+    if (parts.size() == 3) {
```

Comment:
if the deployment name has '-' then this will not work. so condition should be at least 3 and strip of last two parts to make it deployment name.

Reply:
Make sense, upated.